clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -9577,9 +9606,23 @@ void MemVarRule::runRule(const MatchFinder::MatchResult &Result) {
                                 buildString("reinterpret_cast<", UO->getType(),
                                             ">(&", Decl->getName(), ")")));
           }
+          ReplacementEmplaced = true;
         }
       }
     }
+    if (!ReplacementEmplaced && Decl->hasAttr<CUDAConstantAttr>() &&
+        (MemVarRef->getType()->getTypeClass() ==
+             Type::TypeClass::ConstantArray ||
+         MemVarRef->getType()->getTypeClass() == Type::TypeClass::Pointer)) {
```

Comment:
This condition can be removed. Test case kernel6 is caused by `&`, it should not be processed in this `if` block

Reply:
done